### PR TITLE
primitive auth support

### PR DESCRIPTION
**Pull request by aluzzardi**
/cc @gerhard @marcosnils

### CODE REVIEW

File: api/auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+	if state != responseState {
+		return fmt.Errorf("corrupted login challenge (%q != %q)", state, responseState)
+	}
+
+	if oauthError := r.URL.Query().Get("error"); oauthError != "" {
+		description := r.URL.Query().Get("error_description")
+		return fmt.Errorf("authentication error: %s (%s)", oauthError, description)
+	}
+
+	token, err := authConfig.Exchange(context.TODO(), r.URL.Query().Get("code"))
```
[the remark this answers is not in the document]
Comment (aluzzardi) [reply api/auth/auth.go:97]: Copy pasta from blocklayer code :)

File: api/auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+			<head>
+			<script>window.close()</script>
+			<body>
+			%s
+			</body>
+			</html>
+			`, message)))
+		requestCh <- r
+	})
+	go http.ListenAndServe(fmt.Sprintf("localhost:%d", callbackPort), nil)
```
[the remark this answers is not in the document]
Comment (aluzzardi) [reply api/auth/auth.go:72]: Done

2:02PM FTL system | auth server failed: listen tcp 127.0.0.1:38932: bind: address already in use

**Review by marcosnils — APPROVED** [review]
LGTM!

**Comment by aluzzardi** [comment]
@marcosnils Should I merge with the test command or not? It's more of an example than anything.

**Comment by marcosnils** [comment]
> @marcosnils Should I merge with the test command or not? It's more of an example than anything.

I think we can remove this since for testing purposes we can move this behavior to the API. I don't see any value in leaving this in the OSS dagger CLI

**Review by gerhard — COMMENTED** [review]
Looking good!

Pushing a few changes so that we can get this merged and continue iterating on it.

**Pull request merged**